### PR TITLE
ci: GHA - Build & Test on PR & merge

### PR DESCRIPTION
**Pull request by karlskewes**
Part of #4 

Part 1 of migrating CI from Gitlab - build and test on PR.

Using our OSS container image build/render jsonnet to json dashboards
and yaml Prometheus alerts & rules.

Then using upstream Prometheus image run `promtool` to test syntax and
alert condition tests.

Part 2 to follow separately - release.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    needs: [build]
+    steps:
+      - name: Download rendered manifests
+        uses: actions/download-artifact@v2
+        with:
+          name: manifests
+
+      - name: Show files
+        run: |
+          find .
```
Comment (benclapp) [inline .github/workflows/ci.yml]: Still need this one?

Comment (benclapp) [reply .github/workflows/ci.yml]: Otherwise lgtm!

Comment (karlskewes) [reply .github/workflows/ci.yml]: Not necessarily but it's helpful if wondering where missing test file(s) are.
This happened with `tests.yaml` when I first added `manifests/` upload step.

Comment (karlskewes) [reply .github/workflows/ci.yml]: Though we can download the manifests artifact from GHA after build anyway. Removed.

**Pull request merged**